### PR TITLE
Scripts/Deadmines: Fix #6624 Mr. Smite's event text, and phase switch…

### LINKED ISSUE #6624: Mr. Smite
Does not state sound emotes between phases ...

At dual-wield supposed to trigger the sound 
You land lubbers are tougher than I thought, I'll have to improvise.

At the phase wielding the hammer, Mr Smite sound emotes ...
Now you're making me angry ...

Found the sounds using MPQ editor, but unsure of how or what code to use to place in script.

TC 5957a2b10d57+ Win 32

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/64629-mr-smite?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by henhouse**
…ing.
- Fixes Mr. Smite announcing to instance once doors have been destroyed.
- Add missing dialogue to encounter phases.
- Now properly runs in front of his chest, kneels down and equips items, stands, and reengages like he's supposed to.
- Fix weapon model for first phase when he switched to double axes.

Closes #6624

### CODE REVIEW

**Comment by henhouse** [comment]
Issue #6624

**Comment by DDuarte** [comment]
Thanks for your PR! :+1: Please rewrite the SQL to match the guidelines at https://trinitycore.atlassian.net/wiki/display/tc/SQL+Guidelines

**Comment by henhouse** [comment]
Whoops, sorry! My first time pull requesting here. I'll fix in a sec.

**Comment by Kittnz** [comment]
:+1: , ill check it out this evening and merge if everything works out

**Comment by Kittnz** [comment]
Can you squash these commits @henhouse ?

**Comment by tkrokli** [comment]
If you are uncertain about how to do it, you can use this shortcut in Git Extensions: keep a copy of your current SQL file outside of the branch (just in case), then right-click the previous commit and select **soft reset** (back to the previous commit). You can then open the commit window and choose the option
- [X] Amend Commit

to make the new commit part of the first one. That is how I usually keep a single commit in my PRs.

**Comment by henhouse** [comment]
I've squashed them into one commit, and updated the SQL date for today (I hope that is alright).

**Comment by tkrokli** [comment]
Looks good to me. Only cosmetic detail I could find was the missing back ticks on `creature_text` in line 2 and 3 and `entry` in line 2 (but that is cosmetic and can be fixed when it gets pushed to source).

File: src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp [hunk 1]
```diff
@@ -70,10 +66,12 @@ class instance_deadmines : public InstanceMapScript
             ObjectGuid DefiasPirate1GUID;
             ObjectGuid DefiasPirate2GUID;
             ObjectGuid DefiasCompanionGUID;
+            ObjectGuid MrSmiteGUID;
 
             uint32 State;
             uint32 CannonBlast_Timer;
             uint32 PiratesDelay_Timer;
+            uint32 SmiteAlarmDelay_Timer;
```
Comment (jackpoz) [inline src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp]: this field is not initialized in the ctor, you should always make sure all fields are initialized

**Pull request merged**